config.org
#+TITLE: Custom Settings

* User Config

Set the user full name and email address to identify you
#+begin_src emacs-lisp
(setq user-full-name "Jeremy Anderson"
      user-mail-address "jeanderson@weedmaps.com")
#+end_src

* Load External Plugins

#+begin_src emacs-lisp


#+end_src

* Editor Settings
** Basic Settings

*** Indentation
Indent with 2 spaces

#+begin_src emacs-lisp
(after! web-mode
  (setq web-mode-code-indent-offset 2)
  (setq web-mode-css-indent-offset 2)
  (setq web-mode-markup-indent-offset 2))

(after! js2-mode
  (setq js-indent-level 2)
  (setq indent-tabs-mode nil))
#+end_src

*** Look and feel
Set the default fonts for Doom

#+begin_src emacs-lisp
(setq doom-font (font-spec :family "MesloLGS NF" :size 16)
      doom-big-font (font-spec :family "MesloLGS NF" :size 24))
#+end_src

Set the default theme for Doom

#+begin_src emacs-lisp
(setq doom-theme 'doom-one)
#+end_src

Better ANSI colors

#+begin_src emacs-lisp
(custom-set-faces
  '(ansi-color-blue ((t (:foreground "RoyalBlue1" :weight bold))))
  '(ansi-color-red ((t (:foreground "firebrick" :weight bold)))))
#+end_src

Enable bold and italic for comments and keywords
#+begin_src emacs-lisp
(after! doom-themes
  (setq doom-themes-enable-bold t
        doom-themes-enable-italic t))
(custom-set-faces!
  '(font-lock-comment-face :slant italic)
  '(font-lock-keyword-face :slant italic))
#+end_src

Show line numbers

#+begin_src emacs-lisp
(setq display-line-numbers-type t)
#+end_src

* Company Mode

Enable company mode for everything except...
#+begin_src emacs-lisp
(setq company-global-modes '(not inf-ruby-mode))
#+end_src

* Lisp Settings

#+begin_src emacs-lisp
;; (load (expand-file-name "~/quicklisp/slime-helper.el"))
;; Replace "sbcl" with the path to your implementation
;; (setq inferior-lisp-program "sbcl")
#+end_src

* Org config
#+begin_src emacs-lisp
;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/org/")
#+end_src

* Projectile config

#+begin_src emacs-lisp
(add-hook 'projectile-mode-hook 'projectile-direnv-export-variables)
#+end_src
* TODO Ruby config

Setup ruby to indent by 2 spaces

#+begin_src emacs-lisp
(defvar ruby-indent-to-hook 2)
(add-hook 'ruby-mode-hook
          (lambda ()
            (setq-local ruby-indent-level ruby-indent-to-hook)))
#+end_src

Configure the language server.

#+begin_src emacs-lisp
(after! lsp-mode
  (setq lsp-auto-guess-root t)
  (setq lsp-solargraph-formatting nil)
  (setq lsp-solargraph-symbols nil)
  (setq lsp-solargraph-folding nil))

#+end_src

Configure RSpec to run in docker.

#+begin_src emacs-lisp
(setq rspec-use-docker-when-possible t)
#+end_src

Set the command to use when running docker.

#+begin_src emacs-lisp
(setq rspec-docker-command "docker-compose run --rm")
#+end_src

Don't prefix the spec file with any path for running specs.

#+begin_src emacs-lisp
(setq rspec-docker-cwd "")
#+end_src

Some fancy hacking to override the default wrapper fn for formatting the command used to run specs in docker.

#+begin_src emacs-lisp
(defun rspec--docker--wrapper-fn (rspec-docker-command rspec-docker-container command)
  "Function for wrapping a command for execution inside a dockerized environment. "
  (format "%s %s \"%s\"" rspec-docker-command rspec-docker-container command))

(setq rspec-docker-wrapper-fn 'rspec--docker--wrapper-fn)
#+end_src

Better go to definition

#+begin_src emacs-lisp
(after! robe
  (set-lookup-handlers! 'ruby-mode
    :definition '(projectile-rails-goto-file-at-point robe-jump)
    :documentation #'robe-doc))

(after! ruby-mode
  (set-lookup-handlers! 'ruby-mode
    :definition '(projectile-rails-goto-file-at-point robe-jump)
    :documentation #'robe-doc))

(after! web-mode
  (set-lookup-handlers! 'web-mode
    :definition '(projectile-rails-goto-file-at-point rails-routes-jump)))
#+end_src

* Treemacs

#+begin_src emacs-lisp
(setq doom-themes-treemacs-theme "doom-colors")
#+end_src

* Misc config

#+begin_src emacs-lisp
;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.
#+end_src


custom-settings.org
#+TITLE: Custom Settings

* User Config

Set the user full name and email address to identify you
#+begin_src emacs-lisp
(setq user-full-name "Jeremy Anderson"
      user-mail-address "jeanderson@weedmaps.com")
#+end_src

* Load External Plugins

#+begin_src emacs-lisp


#+end_src

* Editor Settings
** Basic Settings
Set default frame size

#+begin_src emacs-lisp
(add-to-list 'default-frame-alist '(height . 60))
(add-to-list 'default-frame-alist '(width . 200))
#+end_src

*** Indentation
Indent with 2 spaces

#+begin_src emacs-lisp
(after! web-mode
  (setq web-mode-code-indent-offset 2)
  (setq web-mode-css-indent-offset 2)
  (setq web-mode-markup-indent-offset 2))

(after! js2-mode
  (setq js-indent-level 2)
  (setq indent-tabs-mode nil))
#+end_src

*** Look and feel
Set the default fonts for Doom

#+begin_src emacs-lisp
(setq doom-font (font-spec :family "MesloLGS NF" :size 14)
      doom-variable-pitch-font (font-spec :family "Source Sans Pro" :size 15)
      doom-big-font (font-spec :family "MesloLGS NF" :size 24))
#+end_src

Set the default theme for Doom

#+begin_src emacs-lisp
(setq doom-theme 'doom-one)
#+end_src

Better ANSI colors

#+begin_src emacs-lisp
(custom-set-faces
  '(ansi-color-blue ((t (:foreground "RoyalBlue1" :weight bold))))
  '(ansi-color-red ((t (:foreground "firebrick" :weight bold)))))
#+end_src

Enable bold and italic for comments and keywords
#+begin_src emacs-lisp
(after! doom-themes
  (setq doom-themes-enable-bold t
        doom-themes-enable-italic t))
(custom-set-faces!
  '(font-lock-comment-face :slant italic)
  '(font-lock-keyword-face :slant italic))
#+end_src

Show line numbers

#+begin_src emacs-lisp
(setq display-line-numbers-type t)
#+end_src

* Company Mode

Enable company mode for everything except...
#+begin_src emacs-lisp
(setq company-global-modes '(not inf-ruby-mode))
#+end_src

* Lisp Settings

#+begin_src emacs-lisp
;; (load (expand-file-name "~/quicklisp/slime-helper.el"))
;; Replace "sbcl" with the path to your implementation
;; (setq inferior-lisp-program "sbcl")
#+end_src

* Org config
#+begin_src emacs-lisp
;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/org/")
#+end_src

* Projectile config

#+begin_src emacs-lisp
(add-hook 'projectile-mode-hook 'projectile-direnv-export-variables)
#+end_src
* TODO Ruby config

Setup ruby to indent by 2 spaces

#+begin_src emacs-lisp
(defvar ruby-indent-to-hook 2)
(add-hook 'ruby-mode-hook
          (lambda ()
            (setq-local ruby-indent-level ruby-indent-to-hook)))
#+end_src

Configure the language server.

#+begin_src emacs-lisp
(after! lsp-mode
  (setq lsp-auto-guess-root t)
  (setq lsp-solargraph-formatting nil)
  (setq lsp-solargraph-symbols nil)
  (setq lsp-solargraph-folding nil))

#+end_src

Configure RSpec to run in docker.

#+begin_src emacs-lisp
(setq rspec-use-docker-when-possible t)
#+end_src

Set the command to use when running docker.

#+begin_src emacs-lisp
(setq rspec-docker-command "docker-compose run --rm")
#+end_src

Don't prefix the spec file with any path for running specs.

#+begin_src emacs-lisp
(setq rspec-docker-cwd "")
#+end_src

Some fancy hacking to override the default wrapper fn for formatting the command used to run specs in docker.

#+begin_src emacs-lisp
(defun rspec--docker--wrapper-fn (rspec-docker-command rspec-docker-container command)
  "Function for wrapping a command for execution inside a dockerized environment. "
  (format "%s %s \"%s\"" rspec-docker-command rspec-docker-container command))

(setq rspec-docker-wrapper-fn 'rspec--docker--wrapper-fn)
#+end_src

Better go to definition

#+begin_src emacs-lisp
(after! robe
  (set-lookup-handlers! 'ruby-mode
    :definition '(projectile-rails-goto-file-at-point robe-jump)
    :documentation #'robe-doc))

(after! ruby-mode
  (set-lookup-handlers! 'ruby-mode
    :definition '(projectile-rails-goto-file-at-point robe-jump)
    :documentation #'robe-doc))

(after! web-mode
  (set-lookup-handlers! 'web-mode
    :definition '(projectile-rails-goto-file-at-point rails-routes-jump)))
#+end_src

* Treemacs

#+begin_src emacs-lisp
(setq doom-themes-treemacs-theme "doom-colors")
#+end_src

* Misc config

#+begin_src emacs-lisp
;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.
#+end_src


config.el
(setq user-full-name "Jeremy Anderson"
      user-mail-address "jeanderson@weedmaps.com")



(after! web-mode
  (setq web-mode-code-indent-offset 2)
  (setq web-mode-css-indent-offset 2)
  (setq web-mode-markup-indent-offset 2))

(after! js2-mode
  (setq js-indent-level 2)
  (setq indent-tabs-mode nil))

(setq doom-font (font-spec :family "MesloLGS NF" :size 16)
      doom-big-font (font-spec :family "MesloLGS NF" :size 24))

(setq doom-theme 'doom-one)

(custom-set-faces
  '(ansi-color-blue ((t (:foreground "RoyalBlue1" :weight bold))))
  '(ansi-color-red ((t (:foreground "firebrick" :weight bold)))))

(after! doom-themes
  (setq doom-themes-enable-bold t
        doom-themes-enable-italic t))
(custom-set-faces!
  '(font-lock-comment-face :slant italic)
  '(font-lock-keyword-face :slant italic))

(setq display-line-numbers-type t)

(setq company-global-modes '(not inf-ruby-mode))

;; (load (expand-file-name "~/quicklisp/slime-helper.el"))
;; Replace "sbcl" with the path to your implementation
;; (setq inferior-lisp-program "sbcl")

;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/org/")

(add-hook 'projectile-mode-hook 'projectile-direnv-export-variables)

(defvar ruby-indent-to-hook 2)
(add-hook 'ruby-mode-hook
          (lambda ()
            (setq-local ruby-indent-level ruby-indent-to-hook)))

(after! lsp-mode
  (setq lsp-auto-guess-root t)
  (setq lsp-solargraph-formatting nil)
  (setq lsp-solargraph-symbols nil)
  (setq lsp-solargraph-folding nil))

(setq rspec-use-docker-when-possible t)

(setq rspec-docker-command "docker-compose run --rm")

(setq rspec-docker-cwd "")

(defun rspec--docker--wrapper-fn (rspec-docker-command rspec-docker-container command)
  "Function for wrapping a command for execution inside a dockerized environment. "
  (format "%s %s \"%s\"" rspec-docker-command rspec-docker-container command))

(setq rspec-docker-wrapper-fn 'rspec--docker--wrapper-fn)

(after! robe
  (set-lookup-handlers! 'ruby-mode
    :definition '(projectile-rails-goto-file-at-point robe-jump)
    :documentation #'robe-doc))

(after! ruby-mode
  (set-lookup-handlers! 'ruby-mode
    :definition '(projectile-rails-goto-file-at-point robe-jump)
    :documentation #'robe-doc))

(after! web-mode
  (set-lookup-handlers! 'web-mode
    :definition '(projectile-rails-goto-file-at-point rails-routes-jump)))

(setq doom-themes-treemacs-theme "doom-colors")

;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.
